lib/resty/healthcheck.lua

```diff
@@ -436,20 +432,47 @@ local function locking_target(self, ip, port, hostname, fn)
   end
   local lock_key = key_for(self.TARGET_LOCK, ip, port, hostname)
 
-  local ok, err = lock:lock(lock_key)
-  if not ok then
-    return nil, "failed to acquire lock: " .. err
+  local pok, perr = pcall(resty_lock.lock, lock, lock_key)
+  if not pok then
+    self:log(DEBUG, "failed to acquire lock: ", perr)
+    return nil, "failed to acquire lock"
   end
 
   local final_ok, final_err = fn()
 
-  ok, err = lock:unlock()
+  local ok, err = lock:unlock()
   if not ok then
     -- recoverable: not returning this error, only logging it
     self:log(ERR, "failed to release lock '", lock_key, "': ", err)
   end
 
   return final_ok, final_err
+
+end
+
+
+-- Run the given function holding a lock on the target.
+-- WARNING: the callback will run unprotected, so it should never
```

Comment:
I don't understand why this comment is needed. What's the advantage of invoking the function in an unprotected way? I'd rather use `pcall` and remove this comment

Reply:
Good call! Fixed, thanks.